src/Analysis/Engine/Impl/Interpreter/Ast/NameLookupContext.cs

```diff
@@ -346,27 +354,61 @@ sealed class NameLookupContext {
             }
 
             var m = GetValueFromExpression(expr.Target);
-            if (m is IPythonType type) {
-                if (type.TypeId == BuiltinTypeId.Type && type == Interpreter.GetBuiltinType(BuiltinTypeId.Type) && expr.Args.Count >= 1) {
-                    var aType = GetTypeFromValue(GetValueFromExpression(expr.Args[0].Expression));
-                    if (aType != null) {
-                        return aType;
+            IMember value = null;
+            switch (m) {
+                case IPythonType type:
+                    if (type.TypeId == BuiltinTypeId.Type && type == Interpreter.GetBuiltinType(BuiltinTypeId.Type) && expr.Args.Count >= 1) {
+                        value = GetTypeFromValue(GetValueFromExpression(expr.Args[0].Expression, options));
+                    } else {
+                        value = new AstPythonConstant(type, GetLoc(expr));
                     }
-                }
-                return new AstPythonConstant(type, GetLoc(expr));
+                    break;
+                default:
+                    value = GetValueFromPropertyOrFunction(m, expr);
+                    break;
             }
+            if (value == null) {
+                _log?.Log(TraceLevel.Verbose, "UnknownCallable", expr.Target.ToCodeString(Ast).Trim());
+                return new AstPythonConstant(_unknownType, GetLoc(expr));
+            }
+            return value;
+        }
+
+        private IMember GetValueFromPropertyOrFunction(IMember fn, Expression expr) {
+            switch (fn) {
+                case IPythonBoundFunction bf:
+                    return GetValueFromFunction(bf.Function as IPythonFunction2, expr);
+                case IPythonFunction2 f:
+                    return GetValueFromFunction(f, expr);
+                case IBuiltinProperty2 p:
+                    return GetPropertyReturnType(p, expr);
+            }
+            return null;
```

Comment:
Shouldn't we return `AstPythonConstant` here?

Reply:
`GetValueFromMember` and `GetValueFromCallable` and `GetValueFromExpression` expect nulls for logging purposes I guess. However, I don't know if there is much value in it.

Reply:
I think point of returning nulls is to reduce amount of 'unknowns' objects. Ex if we return unknown constant it gets stuck in return types while null is skipped by `ExcludeDefault` clause

Reply:
Perhaps we should stick with nulls to save some memory?

Reply:
I am actually cleaning up and returning more nulls now.

Reply:
Ah, so that is intentional. Then ok.

Reply:
Well, it got worse. `AstPythonMultipleMembers.Combine` does not accept two null while it is actually possible. For safety reasons I will stick with unknowns for now.